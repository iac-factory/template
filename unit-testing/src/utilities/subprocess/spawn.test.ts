import { initialize } from ".";

describe( "Subprocess", () => {
    initialize();

    it( "Spawn Class Usage", async () => {
        expect.assertions( 2 );

        const { Spawn } = await import("..");

        const result = await new Spawn( {
            application: "echo", parameters: [
                "Unit Testing ..."
            ], prefix: "Class"
        } );

        const snapshot = JSON.stringify(result, null, 4);

        const state: import("Unit-Testing").State = {
            ... expect.getState(), ... {
                data: { result, snapshot }
            }
        };

        expect.setState( state );

        expect( result ).toBeTruthy();
        expect(snapshot).toMatchSnapshot();
    } );

    it( "Spawn Function Usage", async () => {
        expect.assertions( 2 );

        const { Spawn } = await import("..");

        const result = await Spawn( {
            application: "echo", parameters: [
                "Unit Testing ..."
            ], prefix: "Callable"
        } );

        const snapshot = JSON.stringify(result, null, 4);

        const state: import("Unit-Testing").State = {
            ... expect.getState(), ... {
                data: { result, snapshot }
            }
        };

        expect.setState( state );

        expect( result ).toBeTruthy();
        expect(snapshot).toMatchSnapshot();
    } );

    it( "Spawn Class Output", async () => {
        expect.assertions( 2 );

        const { Spawn } = await import("..");

        const result = await new Spawn( {
            application: "echo", parameters: [
                "Unit Testing ..."
            ], prefix: "Class"
        } );

        const snapshot = JSON.stringify(result, null, 4);

        const state: import("Unit-Testing").State = {
            ... expect.getState(), ... {
                data: { result, snapshot }
            }
        };

        expect.setState( state );

        expect( result ).toEqual(["Unit Testing ..."]);
        expect(snapshot).toMatchSnapshot();
    } );

    it( "Spawn Function Output", async () => {
        expect.assertions( 2 );

        const { Spawn } = await import("..");

        const result = await Spawn( {
            application: "echo", parameters: [
                "Unit Testing ..."
            ], prefix: "Callable"
        } );

        const snapshot = JSON.stringify(result, null, 4);

        const state: import("Unit-Testing").State = {
            ... expect.getState(), ... {
                data: { result, snapshot }
            }
        };

        expect.setState( state );

        expect( result ).toEqual(["Unit Testing ..."]);
        expect(snapshot).toMatchSnapshot();
    } );
} );
