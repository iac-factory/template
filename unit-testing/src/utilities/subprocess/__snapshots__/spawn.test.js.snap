// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subprocess Spawn Class Output 1`] = `
"[
    \\"Unit Testing ...\\"
]"
`;

exports[`Subprocess Spawn Class Usage 1`] = `
"[
    \\"Unit Testing ...\\"
]"
`;

exports[`Subprocess Spawn Function Output 1`] = `
"[
    \\"Unit Testing ...\\"
]"
`;

exports[`Subprocess Spawn Function Usage 1`] = `
"[
    \\"Unit Testing ...\\"
]"
`;
